src/test/migrations/steps/20241114204645-step-fix-datatable-grid-layouts-and-taxon-codes.ts

```diff
@@ -482,7 +482,53 @@ const _fixPanEuropean2025GridLayouts = async (client: BaseProtocol) => {
   await AssessmentController.generateMetadataCache({ assessment }, client)
 }
 
+const _fixTaxonCodes = async (client: BaseProtocol) => {
+  const assessments = await AssessmentController.getAll({}, client)
+
+  /* eslint-disable no-useless-escape */
+  await Promise.all(
+    assessments.map((assessment) =>
+      Promise.all(
+        assessment.cycles.map((cycle) => {
+          const schemaCycle = Schemas.getNameCycle(assessment, cycle)
+          return client.query(`
+            with nodes_to_update as (
+              select
+                  n.id,
+                  n.value,
+                  t.code as matching_taxon_code,
+                  t.scientific_name as original_scientific_name
+              from ${schemaCycle}.node n
+              join ext_data.taxon t
+                  on lower(trim(both ' ' from regexp_replace(n.value->>'raw', '\s+', ' ', 'g'))) = lower(t.scientific_name)
+              join assessment_fra.col c
+                  on c.uuid = n.col_uuid and c.props->>'colType' = 'taxon'
+              where n.value->>'taxonCode' is null
+            ),
+            updated_values as (
+              select
+                id,
+                jsonb_set(
+                  jsonb_set(value, '{taxonCode}', to_jsonb(matching_taxon_code::text), true),
+                  '{raw}',
+                  to_jsonb(original_scientific_name),
+                  true
+                ) as new_value
+              from nodes_to_update
+            )
+            update ${schemaCycle}.node
+            set value = updated_values.new_value
+            from updated_values
+            where ${schemaCycle}.node.id = updated_values.id;
+          `)
+        })
+      )
+    )
+  )
+}
+
```

Comment:
The regexp_replace takes care of cases like:
![image](https://github.com/user-attachments/assets/fd8cb9ce-1487-45e4-a175-50436dc44013)
![2](https://github.com/user-attachments/assets/94fe44aa-dfb4-4fd4-afb4-baa7a81f7408)
![Pasted Graphic 4](https://github.com/user-attachments/assets/83578ec8-6d37-44bb-9093-deb1c475f010)